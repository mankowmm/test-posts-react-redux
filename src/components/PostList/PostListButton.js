import React, { Component } from 'react';
import { Link } from 'react-router-dom'

export class PostListButton extends Component {
    render() {
        return <Link className="btn btn-secondary" to={{pathname: `/post/${this.props.post.id}`}}>See detail</Link>
    }
}

