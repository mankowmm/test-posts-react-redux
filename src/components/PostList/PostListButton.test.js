import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import { shallow } from 'enzyme';
import { PostListButton } from './PostListButton';
import { MemoryRouter } from 'react-router-dom';

describe('PostListButton component test suite', ()=> {

    let elWrapper;
    let postMock;

    beforeEach(() => {
        postMock = {id: 1, title: 'POSTMOCK'};
    })
    
    it('should validate with snapshot', () => {
        const tree = renderer
          .create(<MemoryRouter><PostListButton post={postMock}/></MemoryRouter>)
          .toJSON();
        expect(tree).toMatchSnapshot();
    });          

});