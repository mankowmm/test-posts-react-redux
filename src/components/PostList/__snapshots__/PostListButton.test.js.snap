// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostListButton component test suite should validate with snapshot 1`] = `
<a
  className="btn btn-secondary"
  href="/post/1"
  onClick={[Function]}
>
  See detail
</a>
`;
